romancal/outlier_detection/tests/test_outlier_detection.py

```diff
@@ -1,42 +1,358 @@
+import json
 import os
+from io import StringIO
+from pathlib import Path
 
+import numpy as np
 import pytest
+from astropy import coordinates as coord
 from astropy import units as u
+from astropy.modeling.models import RotationSequence3D, Scale, Shift
+from gwcs import coordinate_frames as cf
+from gwcs import wcs
+from gwcs.geometry import CartesianToSpherical, SphericalToCartesian
+from roman_datamodels import datamodels as rdm
 from roman_datamodels import maker_utils
-from roman_datamodels.datamodels import ImageModel
 
-from romancal.outlier_detection import OutlierDetectionStep
+from romancal.datamodels import ModelContainer
+from romancal.outlier_detection import OutlierDetectionStep, outlier_detection
 
 
-@pytest.mark.skipif(
-    os.environ.get("CI") == "true",
-    reason=(
-        "Roman CRDS servers are not currently available outside the internal network"
-    ),
+def _create_tel2sky_model(input_dm):
+    """
+    Create the transform from telescope to sky.
+
+    The transform is defined with a reference point in a Frame
+    associated with the telescope (V2, V3) in arcsec, the corresponding
+    reference point on sky (RA_REF, DEC_REF) in deg, and the position angle
+    at the center of the aperture, ROLL_REF in deg.
+
+    Parameters
+    ----------
+    input_dm : roman_datamodels.ImageModel
+        A Roman image datamodel.
+
+    Returns
+    -------
+    astropy.modeling.core.CompoundModel
+        An astropy compound model with the transformations to map
+        the telescope reference system onto the world reference systems.
+    """
+    v2_ref = input_dm.meta.wcsinfo.v2_ref / 3600
+    v3_ref = input_dm.meta.wcsinfo.v3_ref / 3600
+    roll_ref = input_dm.meta.wcsinfo.roll_ref
+    ra_ref = input_dm.meta.wcsinfo.ra_ref
+    dec_ref = input_dm.meta.wcsinfo.dec_ref
+
+    angles = np.array([v2_ref, -v3_ref, roll_ref, dec_ref, -ra_ref])
+    axes = "zyxyz"
+    rot = RotationSequence3D(angles, axes_order=axes)
+
+    # The sky rotation expects values in deg.
+    # This should be removed when models work with quantities.
+    model = (
+        (Scale(0.1 / 3600) & Scale(0.1 / 3600))
+        | SphericalToCartesian(wrap_lon_at=180)
+        | rot
+        | CartesianToSpherical(wrap_lon_at=360)
+    )
+    model.name = "v23tosky"
+    return model
```

Comment:
Why the copy/duplicaiton of this function? Can you call the one in [pointing](https://github.com/spacetelescope/romancal/blob/main/romancal/assign_wcs/pointing.py#L6)?